test/ibmq/test_experiment.py

```diff
@@ -235,6 +235,47 @@ def test_experiments_with_hgp_error(self):
                 for key in missing_keys:
                     self.assertIn(key, str(ex_cm.exception))
 
+    def test_experiments_with_public_filters(self):
```

Comment:
Since we're really testing 2 separate keywords (`exclude_public` and `public_only`), it's probably better to split the test case into 2, or use `subTest`.